src/runtime/manifest.ts

```diff
@@ -1315,13 +1317,26 @@ ${e.message}
   registerStore(store: UnifiedStore, tags: string[]): void {
     // Only register stores that have non-volatile storage key and don't have a
     // #volatile tag.
-    if (!this.findStoreById(store.id) &&
-        (store.storageKey as StorageKey).protocol !== 'volatile' &&
-        !tags.includes('volatile')) {
+    if (!this.findStoreById(store.id) && !this.isVolatileStore(store, tags)) {
+      if (Flags.defaultReferenceMode && !(store.storageKey instanceof ReferenceModeStorageKey)) {
+        // Note: skip adding backing and container stores to context, if using reference-mode stores.
+        // TODO: there must be a better way!
+        return;
+      }
       this._addStore(store, tags);
     }
   }
 
+  isVolatileStore(store: UnifiedStore, tags: string[]): boolean {
+    if ((store.storageKey as StorageKey).protocol === VolatileStorageKey.protocol) return true;
+    if ((store.storageKey as StorageKey).protocol === ReferenceModeStorageKey.protocol &&
+        (store.storageKey as ReferenceModeStorageKey).storageKey.protocol === VolatileStorageKey.protocol) {
```

Comment:
might want to check both keys in the ReferenceModeStorageKey here.

Reply:
done